src/main/kotlin/org/neo4j/graphql/handler/filter/OptimizedFilterHandler.kt

```diff
@@ -340,8 +359,16 @@ class OptimizedFilterHandler(val type: GraphQLFieldsContainer) {
                         FieldOperator.values()
                             .map { it to definedField.name + it.suffix }
                             .mapNotNull { (predicate, queryFieldName) ->
-                                queriedFields.remove(queryFieldName)?.let { (index, objectField) ->
-                                    Predicate(predicate, objectField, definedField, index)
+                                queriedFields[queryFieldName]?.let { (index, objectField) ->
+                                    if (predicate.requireParam xor (objectField.value !is NullValue)) {
```

Comment:
first time I've seen an XOR in real code. esp. xor with negations is kinda ouch

Reply:
:-)